### ISSUE TITLE
Auth methods fetch doesn't use Horizon host

### ISSUE THREAD

**Issue by mlynch** [rethinkdb/horizon#422]
Server version: 1.0.1
Client version: (loaded from horizon server, 1.0.1?)

Congrats on the launch, I'm _so_ excited about this project!

I'm running Horizon as a standalone server and then running a static app on another server. That all works great, except the auth system (trying the token auth with Twitter) is incorrectly trying to fetch the horizon endpoint info from my static server instead of the horizon server.

I checked my `/horizon/auth_methods` endpoint on the horizon server and I get back `{"twitter":"/horizon/twitter"}` so it seems properly configured.

My login code looks like this:

``` javascript
horizon.authEndpoint('twitter').toPromise().then((endpoint) => {
  window.location.pathname = endpoint;
})
```

When this is called, Horizon tries to do a fetch for the auth methods from the server, but it doesn't use the proper origin, instead using the origin of my static server which is not where I'm running Horizon (and never will be, since ultimately it will be loaded in a mobile app).

This results, predictably, in a 404.  Here's the failing line: https://github.com/rethinkdb/horizon/blob/061f784fa6f8cd6b30ffcaf6b88e823e0e386b3f/client/src/auth.js#L19

(also perhaps that console.log should go as well)

**Comment by danielmewes** [comment]
Looks like a bug / oversight. Thanks for reporting it.

**Comment by deontologician** [comment]
We should probably allow customizing this redirect somehow. @Tryneus is there a way to specify the full url to redirect to in the auth provider configuration?

**Comment by deontologician** [comment]
OK! I think this is actually solveable. The config.toml has a setting for the redirect url:

``` toml
auth_redirect = "/"
```

By default, it's the root of the server the js file is served from, but you can put in the url of your horizon server.

@mlynch let me know if this works for you

**Comment by mlynch** [comment]
@deontologician cool, ran into this issue (submitted a PR for it) where auth_redirect can't be specified in config.toml as it was part of the yes_no_options list: https://github.com/rethinkdb/horizon/pull/439

**Comment by mlynch** [comment]
Okay, just pulled locally with my change, and it fixes the config issue but the `/horizon/auth_methods` fetch is still using my static host origin.

**Comment by deontologician** [comment]
Ah ha! Ok, we need another config option for that. I am going to metamorphose this issue for that purpose. Thanks for looking into it @mlynch !

**Comment by gutiNapo** [comment]
Hello @deontologician ,

There is no need of a new configuration: **horizon.authEndpoint** should use the host defined when creating Horizon object (ex: var horizon = Horizon({**host**: '192.168.99.100:8181', authType: 'token'});). 

At https://github.com/rethinkdb/horizon/blob/061f784fa6f8cd6b30ffcaf6b88e823e0e386b3f/client/src/auth.js#L19, the **fetchJSON** should be adapted to use the host.
Something as `return fetchJSON('**${this._host}/**${this._horizonPath}/auth_methods')` instead of just `fetchJSON('${this._horizonPath}/auth_methods')`

Maybe the easiest way to avoid other similar problems would be to adapt **_horizonPath** directly at https://github.com/rethinkdb/horizon/blob/061f784fa6f8cd6b30ffcaf6b88e823e0e386b3f/client/src/index.js#L97 to include the defined host (as `horizon._horizonPath = host + '/' + path`).

Ps: Just started with Horizon and it looks really promising

**Comment by deontologician** [comment]
@dlecan also reported this on the forums: https://discuss.horizon.io/t/horizon-client-doesnt-seem-to-use-host-and-secure-properties/286

**Comment by dlecan** [comment]
Yes, I have the same issue.
As a temporary workaround, I configured the webpack-dev-middleware to proxy all requests.

**Comment by deontologician** [comment]
Ok, fix for this finally up in #642

**Comment by luandro** [comment]
I'm running into this, is it really fixed? In #561 @danielmewes stated it would only be fixed in upcoming 2.0.0 release. Shouldn't this issue remain open until fixed?  Is there no way to have a front-end app running in a separate server from horizon?
